src/typing/constraint_js.ml

```diff
@@ -507,7 +507,16 @@ type block_entry = {
   def_loc: Spider_monkey_ast.Loc.t option;
   for_type: bool;
 }
-type block = block_entry SMap.t ref
+
+type block_scope =
+  | Hoist (* var, class, functions hoisted up to this point *)
+  | Block (* let, const *)
+
+type block = {
```

Comment:
Now that 'block' is term that refers to a kind of scope (which isn't the same as what this name refers to as a "block"), it's probably time to rename this structure to 'scope'